src/Repository/ProductRepositoryTrait.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusAlgoliaPlugin\Repository;
+
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+use Setono\SyliusAlgoliaPlugin\DTO\ProductIndexScope;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+
+/**
+ * @mixin EntityRepository
+ */
+trait ProductRepositoryTrait
+{
+    public function createQueryBuilderFromProductIndexScope(ProductIndexScope $productIndexScope): QueryBuilder
+    {
+        // todo recheck this query
```

Comment:
I am not the biggest Doctrine/SQL expert so I am not 100% sure this query is what we intend it to be

Reply:
I would just avoid chaining calls, as it is not the best approach. But other than that, it seems to be correct

Reply:
All I chain are query builder methods. Everything else I don't :D

Reply:
Here's an article on it: https://ocramius.github.io/blog/fluent-interfaces-are-evil/

Reply:
Of course he will say fluent is good on query builder... He is biased on this :D 

IMO it is not a good approach as any change in the middle that introduce bigger logic will result in git changelog being messed up...

Same for trailing coma in the end of PHP array, or now in method arguments

Reply:
:D